tubular/dates.py

```diff
@@ -1033,37 +1023,17 @@ def _map_values(self, value: float, interval: str) -> str:
         str : str
             Mapped value
         """
-        if type(value) is not float and type(value) is not int:
-            msg = f"{self.classname()}: value should be float or int"
-            raise TypeError(msg)
-
-        errors = {
-            "timeofday": "0-23",
-            "dayofweek": "0-6",
-            "timeofmonth": "1-31",
-            "timeofyear": "1-12",
-        }
-        ranges = {
-            "timeofday": (0, 24, 1),
-            "dayofweek": (0, 7, 1),
-            "timeofmonth": (1, 32, 1),
-            "timeofyear": (1, 13, 1),
-        }
-        mappings = {
-            "timeofday": self.timeofday_mapping,
-            "dayofweek": self.dayofweek_mapping,
-            "timeofmonth": self.timeofmonth_mapping,
-            "timeofyear": self.timeofyear_mapping,
-        }
-
-        if (not np.isnan(value)) and (value not in np.arange(*ranges[interval])):
-            msg = f"{self.classname()}: value for {interval} mapping  in self._map_values should be an integer value in {errors[interval]}"
-            raise ValueError(msg)
+        if isinstance(value, float):
+            if np.isnan(value):
+                return np.nan
+            if value.is_integer():
+                value = int(value)
 
-        if np.isnan(value):
-            return np.nan
+        if isinstance(value, int) and value in self.RANGE_TO_MAP[include_option]:
+            return self.inverted_datetime_mappings[include_option][value]
 
-        return mappings[interval][value]
+        msg = f"{self.classname()}: value for {include_option} mapping in self._map_values should be an integer value in {min(self.RANGE_TO_MAP[include_option])}-{max(self.RANGE_TO_MAP[include_option])}"
```

Comment:
S/N - feel like this might read better as an `else` even if not required for the logic to work.

Reply:
personally I don't disagree, but this has been beaten into me by ruff rules that we've added to HomeX haha https://docs.astral.sh/ruff/rules/superfluous-else-raise/

If we think we might turn on more of these in future maybe would leave?

Reply:
I bow to the authority of ruff in that case.  Prefer to leave stylistic choices to a linter where possible.